rqt_gui_cpp/src/rqt_gui_cpp/nodelet_plugin_provider.h

```diff
@@ -86,6 +87,9 @@ class NodeletPluginProvider
     virtual ~RosSpinThread();
     void run();
     bool abort;
+    // Create an executor that will be responsible for execution of callbacks for a set of nodes.
+    // With this version, all callbacks will be called from within this thread (the main one).
+    rclcpp::executors::SingleThreadedExecutor exec_;
```

Comment:
I chose to use a MultiThreadedExecutor on the python equivalent. It might help plugins from not blocking each other.